subsys/bluetooth/audio/mcs.c

```diff
@@ -909,120 +1130,170 @@ static void mark_icon_url_changed_cb(struct bt_conn *conn, void *data)
 		return;
 	}
 
-	client->icon_url_changed = true;
+	atomic_set_bit(client->flags, POINTER_TO_UINT(data));
+	k_work_submit(&deferred_nfy_work);
 }
 
-void media_proxy_sctrl_icon_url_cb(const char *name)
+static void media_proxy_sctrl_player_name_cb(const char *name)
 {
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_PLAYER_NAME_CHANGED));
+}
 
+void media_proxy_sctrl_icon_url_cb(const char *name)
+{
 	bt_conn_foreach(BT_CONN_TYPE_LE, mark_icon_url_changed_cb, NULL);
 }
 
 void media_proxy_sctrl_track_changed_cb(void)
 {
-	LOG_DBG("Notifying track change");
-	notify(BT_UUID_MCS_TRACK_CHANGED, NULL, 0);
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_TRACK_CHANGED));
 }
 
 void media_proxy_sctrl_track_title_cb(const char *title)
 {
-	LOG_DBG("Notifying track title: %s", title);
-	notify_string(BT_UUID_MCS_TRACK_TITLE, title);
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_TRACK_TITLE_CHANGED));
 }
 
 void media_proxy_sctrl_track_position_cb(int32_t position)
 {
-	LOG_DBG("Notifying track position: %d", position);
-	notify(BT_UUID_MCS_TRACK_POSITION, &position, sizeof(position));
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_TRACK_POSITION_CHANGED));
 }
 
 void media_proxy_sctrl_track_duration_cb(int32_t duration)
 {
-	LOG_DBG("Notifying track duration: %d", duration);
-	notify(BT_UUID_MCS_TRACK_DURATION, &duration, sizeof(duration));
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_TRACK_DURATION_CHANGED));
 }
 
 void media_proxy_sctrl_playback_speed_cb(int8_t speed)
 {
-	LOG_DBG("Notifying playback speed: %d", speed);
-	notify(BT_UUID_MCS_PLAYBACK_SPEED, &speed, sizeof(speed));
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_PLAYBACK_SPEED_CHANGED));
 }
 
 void media_proxy_sctrl_seeking_speed_cb(int8_t speed)
 {
-	LOG_DBG("Notifying seeking speed: %d", speed);
-	notify(BT_UUID_MCS_SEEKING_SPEED, &speed, sizeof(speed));
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_SEEKING_SPEED_CHANGED));
 }
 
+#if defined(CONFIG_BT_OTS)
 void media_proxy_sctrl_current_track_id_cb(uint64_t id)
 {
-	LOG_DBG_OBJ_ID("Notifying current track ID: ", id);
-	notify(BT_UUID_MCS_CURRENT_TRACK_OBJ_ID, &id, BT_OTS_OBJ_ID_SIZE);
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_CURRENT_TRACK_OBJ_ID_CHANGED));
 }
```

Comment:
For all these functions where the value is no longer directly notified, we can remove the arguments, right? i.e. `id` here is now unused

Reply:
True, but those can be removed in the future. The callbacks itself can be removed to use just only one callback with flags indicating which values changed.